### PR TITLE
Update frozen upstream requirements

### PR DESCRIPTION
**Pull request by johnwalker**
*Issue #, if available:*

*Description of changes:*

This updates the upstream requirements and documents how to do it.

I wanted to document how the upstream requirements are used, but I didn't know. What are they used for?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mattsb42-aws** [comment]
@johnwalker That's a good point; I think it's currently "documented" in a combination of code and commit comments.

The backstory here is that pyca/cryptography runs downstream tests as part of their CI to make sure that their changes do not break certain important downstream dependencies. This client and the AWS Encryption SDK for Python are among those that they test.

We had multiple cases where their CI broke because their downstream tests of our clients were broken because of new versions of one of *our* dependencies. This is something that *we* want to be aware of (and our CI is set up to catch this), but it is ***not*** something that we want an upstream dependency like pyca/cryptography to break on.

To address this issue, I set up these frozen upstream dependency definitions to freeze the versions of *our* dependencies that will be used in any tests run by upstream dependencies[1]. We also run these isolated tests in our own CI to make sure that changes to the frozen dependencies do not break those upstream tests.

[1] https://github.com/pyca/cryptography/blob/master/.travis/downstream.d/dynamodb-encryption-sdk.sh

**Comment by johnwalker** [comment]
Thank you! I updated the readme to include explanation.

File: test/README.rst [hunk 1]
```diff
@@ -14,3 +14,24 @@ In order to run these integration tests successfully, these things which must be
 
 .. _automatically discoverable credential locations: http://boto3.readthedocs.io/en/latest/guide/configuration.html
 .. _AWS KMS CMK ARN: http://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html
+
+Updating Upstream Requirements
+==============================
+
+The purpose of the upstream requirements files is to provide a stable list of
+packages for pyca/cryptography's downstream test of the DynamoDB Encryption
```
Comment (mattsb42-aws) [inline test/README.rst]: I deliberately generalized this when I was building the infrastructure, and I think this should be reflected in the description. These are for any upstream dependency that wants to test their downstream consumers. It just so happens that currently pyca/cryptography is the only instance of this.

Comment (johnwalker) [reply test/README.rst]: gotcha, fixed

**Pull request merged**